From: Chad Monroe <chad@monroe.io>
Date: Wed, 25 Feb 2026 19:07:10 -0800
Subject: [PATCH] wifi-scripts: disable background radar in AP+STA mode

When an AP is colocated with a supplicant interface the AP must follow
that interface's channel. Background radar CAC can trigger a channel
switch to a DFS-cleared channel the supplicant can't follow, causing
tear down of colocated VAPs.

The wireless handler already flags this case: mac80211.sh setup() sets
data.channel_follow for sta, mesh and adhoc supplicant interfaces alike,
which is the same condition hostapd uses for channel_follow and the same
one that drives ap_start_disabled. Carry it into the device config as
has_supplicant and skip enable_background_radar on it in the device
config generator (usr/share/ucode/wifi/hostapd.uc).

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -524,7 +524,7 @@ function generate(config) {
 	append_vars(config, [ 'rssi_reject_assoc_rssi', 'rssi_reject_assoc_timeout', 'rssi_ignore_probe_request', 'iface_max_num_sta' ]);
 
 	/* ACS / Radar*/
-	if (!phy_features.radar_background || config.band != '5g')
+	if (!phy_features.radar_background || config.band != '5g' || config.has_supplicant)
 		delete config.enable_background_radar;
 	else
 		set_default(config, 'enable_background_radar', false);
@@ -582,6 +582,8 @@ export function setup(data) {
 
 	data.config.phy = data.phy;
 
+	data.config.has_supplicant = !!data.channel_follow;
+
 	generate(data.config);
 
 	if (data.config.num_global_macaddr)
